static/app/views/dashboards/manage/index.tsx

```diff
@@ -71,17 +84,85 @@ function ManageDashboards() {
         query: {
           ...pick(location.query, ['cursor', 'query']),
           sort: getActiveSort().value,
-          per_page: '9',
+          per_page: rows * columns,
         },
       },
     ],
     {staleTime: 0}
   );
 
-  const [showTemplates, setShowTemplatesLocal] = useLocalStorageState(
-    SHOW_TEMPLATES_KEY,
-    shouldShowTemplates()
-  );
+  const dashboardsPageLinks = getResponseHeader?.('Link') ?? '';
+
+  useEffect(() => {
+    const dashboardGridObserver = new ResizeObserver(
+      debounce(entries => {
+        entries.forEach(entry => {
+          const currentWidth = entry.contentRect.width;
+
+          setGridWidth(prevWidth => {
+            if (currentWidth !== prevWidth && prevWidth !== 0) {
+              setResizing(true);
+              return currentWidth;
+            }
+            return prevWidth;
+          });
+
+          setRowsAndColumns(currentWidth);
+
+          const paginationObject = parseLinkHeader(dashboardsPageLinks);
+          if (
+            dashboards?.length &&
+            paginationObject.next.results &&
+            rows * columns > dashboards.length
+          ) {
+            refetchDashboards();
+          }
+        });
+      }, 10)
+    );
+
+    if (dashboardGridRef.current) {
+      dashboardGridObserver.observe(dashboardGridRef.current);
+    }
+
+    return () => {
+      if (dashboardGridRef.current) {
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+        dashboardGridObserver.unobserve(dashboardGridRef.current);
+      }
+      setResizing(false);
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Ignoring exhaustive deps leads to the dark side. What dependencies does it want?

Reply:
it wants `[columns, dashboards?.length, dashboardsPageLinks, refetchDashboards, rows]` but i remember that causing it to constantly call the useEffect

Reply:
nvm found a fix 👍